商品購入機能実装

koichiro1007 opened the pull request:
### What
商品購入機能のクレジットカード追加機能、購入機能を実装したのでコードレビューをお願いします。

### Why
商品購入機能実装するため。


クレジットカード追加画面（/card/new）
クレジットカード削除画面（/card/show）
https://gyazo.com/c5b99d1d6fdf6256bc534d1aab96b5e9

商品詳細画面（/items/1） →購入確認画面（purchase/1）→完了画面
https://gyazo.com/bb04f51904c2a7ef5e01ea517cf8fc05
https://gyazo.com/13ee9d305cf6623594f124e0de99d72a

koichiro1007 committed:
商品購入機能実装

クレジットカード機能、商品購入機能実装

TOMARUMARU commented on .env:
```diff
@@ -0,0 +1,2 @@
+PAYJP_PRIVATE_KEY = 'sk_test_5da731fc8cfd2337a1058a3e'
+PAYJP_KEY = 'pk_test_ff42c4dad25c6de95ce40bc7'
```
こちらのファイルはgithubにあげないようにしましょう

TOMARUMARU commented on app/controllers/card_controller.rb:
```diff
@@ -0,0 +1,52 @@
+  end
+
+  def delete #PayjpとCardデータベースを削除します
+    card = Card.where(user_id: current_user.id).first
```
set_cardメソッドを作りましょう

TOMARUMARU commented on app/controllers/purchase_controller.rb:
```diff
@@ -0,0 +1,34 @@
+  end
+
+  def pay
+    item = Item.find(params[:id])
```
set_itemメソッドを作りましょう

TOMARUMARU commented on app/controllers/purchase_controller.rb:
```diff
@@ -0,0 +1,34 @@
+
+  def pay
+    item = Item.find(params[:id])
+    card = Card.where(user_id: current_user.id).first
```
set_cardメソッドを作りましょう

koichiro1007 committed:
修正しました

set_card、set_itemメソッドを作成し、gitignoreに/.envを追記

koichiro1007 committed:
.envファイルを削除

shingoteshima commented:
コンフリクト解消しましょう。

koichiro1007 committed:
Merge branch 'master' into feat/create-credit2

i-date commented on app/controllers/card_controller.rb:
```diff
@@ -0,0 +1,58 @@
+
+  def set_card
+    Card.where(user_id: current_user.id).first
+  end
```
53-55行目は取得したcardデータを変数に入れるようにして、before_actionを使用するようにしましょう。
```
def set_card
  @card = Card.where(user_id: current_user.id).first
end
```

i-date commented on app/controllers/card_controller.rb:
```diff
@@ -0,0 +1,58 @@
+
+  def delete #PayjpとCardデータベースを削除します
+    if set_card.blank?
+    else
```
trueの場合の処理がないので、条件を反転してelseを使用せずに済むようにしましょう。

i-date commented on app/controllers/purchase_controller.rb:
```diff
@@ -0,0 +1,45 @@
+    Payjp::Charge.create(
+    :amount => set_item.price, #支払金額を入力（itemテーブル等に紐づけても良い）
+    :customer => set_card.customer_id, #顧客ID
+    :currency => 'jpy', #日本円
```
24-26行目のハッシュロケットをシンボルに修正しましょう。

i-date commented on app/controllers/purchase_controller.rb:
```diff
@@ -0,0 +1,45 @@
+
+  def set_card
+    Card.where(user_id: current_user.id).first
+  end
```
こちらのset_itemやset_cardも取得したデータを変数に代入するようにして、before_actionを使用しましょう。

i-date reviewed (commented):
追加コメントいたしました。

koichiro1007 committed:
修正しました

@itemと@cardの変数をプライベートメソッドで定義し、before_actionでアクションを起こせるように変更しました。
また、ハッシュロケット 部分はシンボル型へ変更しました。

i-date reviewed (approved):
修正ありがとうございます！LGTMとさせていただきます。
